pymeasure/experiment/results.py

```diff
@@ -104,22 +140,31 @@ class Results(object):
     :param procedure: Procedure object
     :param data_filename: The data filename where the data is or should be
                           stored
+    :param output_format: Formatter which converts the emitted result data
+                          so it can be written to file, defaults to CSV
     """
 
     COMMENT = '#'
     DELIMITER = ','
     LINE_BREAK = "\n"
     CHUNK_SIZE = 1000
 
-    def __init__(self, procedure, data_filename):
+    def __init__(self, procedure, data_filename, output_format='CSV'):
         if not isinstance(procedure, Procedure):
             raise ValueError("Results require a Procedure object")
         self.procedure = procedure
         self.procedure_class = procedure.__class__
         self.parameters = procedure.parameter_objects()
         self._header_count = -1
 
-        self.formatter = CSVFormatter(columns=self.procedure.DATA_COLUMNS)
+        if output_format == 'CSV_PANDAS':
+            self.formatter = CSVFormatter_Pandas(
+                columns=self.procedure.DATA_COLUMNS,
+                delimiter=self.DELIMITER,
+                line_break=self.LINE_BREAK
+            )
+        else:  # default to CSV
+            self.formatter = CSVFormatter(columns=self.procedure.DATA_COLUMNS)
```

Comment:
maybe it would be nicer if instantiating the formatter could be done in a more uniform way (to prevent clutter, especially _if_ in the future new formatters pop up); maybe something like this:
```python
def __init__(.... output_format='CSV'):
    ...
    ...
    formatter_class = {
        'CSV': CSVFormatter,
        'CSV_PANDAS': CSVFormatter_Pandas,
    }[output_format]
    self.formatter = formatter_class(.....)
```
Here the arguments for the formatter-classes need to be uniform; an abstract `Formatter` class might help for that, see the general remark for that.

Reply:
I followed your suggestion. If/When new formatters come up, one probably needs to adapt this anyway...